In [1]:
from __future__ import print_function
import pandas as pd
import pylab as pl
import os

%pylab inline

Populating the interactive namespace from numpy and matplotlib


Question: Is the new bus route improving commute time?

Null Hypothesis H0: the commuting time is longer than or the same with the new bus route as it was before

Alternative Hypothesis Ha: the commuting time is shorter with the new bus route than it was before

H0 = Tn >= To

Ha = Tn < To

Statistical Significance: p-value =.05

Information given: To mean = 36, standard deviation = 6
Sample size is 100

In [2]:
if not os.path.isfile('times.txt'):
    !curl -O https://raw.githubusercontent.com/fedhere/PUI2016_fb55/master/Lab3_fb55/times.txt
pd.read_csv('times.txt')

31.62223931
0     32.821376
1     30.229101
2     31.413766
3     39.010550
4     34.822079
5     39.871884
6     39.579946
7     31.026587
8     27.662461
9     18.875288
10    46.231373
11    30.015226
12    47.600410
13    16.527267
14    42.446233
15    33.625274
16    42.284626
17    46.205207
18    30.875174
19    32.948003
20    26.176915
21    31.256996
22    35.336446
23    35.640725
24    40.322972
25    38.043486
26    29.457979
27    23.916364
28    34.564035
29    45.427825
..          ...
69    31.481028
70    37.149151
71    38.991504
72    20.530594
73    27.804477
74    38.699333
75    30.022314
76    38.648842
77    26.406689
78    32.102250
79    28.315328
80    43.261959
81    40.360941
82    37.302526
83    31.451102
84    38.795995
85    36.925534
86    32.371679
87    23.258356
88    26.426133
89    48.749877
90    31.425124
91    46.806793
92    30.597486
93    20.792632
94    26.197852
95    42.727410
96    33.731470
97    37.238591
98    36.339841

[99 rows x 1 columns]

In [3]:
mtadata = pd.read_csv('times.txt')

In [4]:
mtadata.mean()

31.62223931    34.494888
dtype: float64

34.49 is the mean of times for Tn

In [5]:
z = (36 - 34.5)/ (6/(100**.5))
## z = population mean - sample mean / standard deviation/square root of the sample size

In [6]:
print (z)

2.5


The results of the z-test show that there is a variance of 2.5 between the sample mean (34.5) and the population mean (36). Due to the fact that our statistical significance was choosen to be .05 and this result is futher than 2 sigmas away from the old mean, we can reject the null hypothesis.